### PR TITLE
Switch to Humble/Garden on 22.04

### PR DESCRIPTION
**Pull request by andermi**
Get tests passing

merge along with:
https://github.com/osrf/buoy_msgs/pull/38

### CODE REVIEW

**Comment by andermi** [comment]
I'm at the point where I can't get it to fail on my machine, but it does in CI...

File: .github/workflows/build-and-test.sh [hunk 1]
```diff
@@ -11,18 +11,24 @@ mkdir -p $COLCON_WS_SRC
...
+                   gz-garden \
                    python3-colcon-common-extensions \
                    python3-rosdep \
                    python3-vcstool \
                    wget
 
 cd $COLCON_WS_SRC
 cp -r $GITHUB_WORKSPACE $COLCON_WS_SRC
-wget https://raw.githubusercontent.com/osrf/buoy_entrypoint/main/buoy_all.yaml
+wget https://raw.githubusercontent.com/osrf/buoy_entrypoint/chapulina/humble_garden/buoy_all.yaml
```
Comment (mabelzhang) [inline .github/workflows/build-and-test.sh:31]: Putting a note here so we remember to change this back to `main` after merging.

**Review by mabelzhang — COMMENTED** [review]
I've finally set up my development Docker environment for Humble+Garden in order to test this, along with the 2 other Humble+Garden PRs (https://github.com/osrf/buoy_msgs/pull/38 and https://github.com/osrf/buoy_entrypoint/pull/22).

To compile, this needs a manual export to `PYTHONPATH` (see my comment to add it to the installation instructions in the buoy_entrypoint PR - https://github.com/osrf/buoy_entrypoint/pull/22#discussion_r1068861868).

I've quickly looked through 26/67 files in this PR so far, just checking to make sure no spurious changes (I've seen that happen with the ign/gz renaming Find+Replace recently!).

I'm guessing the testing for this project is done through these two commands:
https://github.com/osrf/buoy_sim/blob/a92b33440ed16a3517b7be5d637fcca8c252e460/.github/workflows/build-and-test.sh#L41-L42
I ran those locally. Test 38 is stuck with a bunch of `/*_command not available`. Do I need to run some of these tests manually?
```
38: Test command: /usr/bin/python3.10 "-u" "/opt/ros/humble/share/ament_cmake_test/cmake/run_test.py" "/home/developer/mbari/buoy_sim_ws/build/buoy_tests/test_results/buoy_tests/launch_no_inputs_py.launch.py.xunit.xml" "--package-name" "buoy_tests" "--output-file" "/home/developer/mbari/buoy_sim_ws/build/buoy_tests/launch_test/launch_no_inputs_py.launch.py.txt" "--command" "/usr/bin/python3" "-m" "launch_testing.launch_test" "/home/developer/mbari/buoy_sim_ws/src/buoy_sim/buoy_tests/launch/no_inputs_py.launch.py" "--junit-xml=/home/developer/mbari/buoy_sim_ws/build/buoy_tests/test_results/buoy_tests/launch_no_inputs_py.launch.py.xunit.xml" "--package-name=buoy_tests"
```

How long has CI been running? It doesn't look like `Build and test (fortress, galactic) Expected — Waiting for status to be reported` ever finished. Oh, just saw the version now, is that something that'll go away after this PR?

**Comment by quarkytale** [comment]
That's right, after we get this is in we can update the CI workflow to not build for galactic and fortress anymore.

**Comment by mabelzhang** [comment]
Notes from group discussion:
Tests pass for Michael A and Dharini locally.
Will disable CI (in another PR, to be merged before this one) until #116 resolved upstream in ROS 2.
Then can get this PR in.

**Review by mabelzhang — COMMENTED** [review]
I've looked at 58/67 files so far.

I'm keeping a list for the reviewer's sanity, in case we need to come back and look at which files have the actual changes to make the tests pass (as opposed to the majority of the files that just have the mindless ign->gz renaming), also in case we need to revert them after disabling CI:
```
buoy_tests/launch/pc_commands_ros_feedback_py.launch.py
buoy_tests/launch/sc_pump_ros_feedback_py.launch.py
buoy_tests/testing_utils/utils.py
buoy_tests/tests/pc_commands_ros_feedback.cpp
buoy_tests/tests/sc_commands_ros_feedback.cpp
buoy_tests/tests/fixture_server.cpp
```
There may be more as I go through the rest of the files, in which case I will edit this list.

**Comment by andermi** [comment]
@mabelzhang @mjcarroll I think it works with cyclonedds?

**Comment by andermi** [comment]
rerunning the jobs

**Comment by mabelzhang** [comment]
The PR test succeeded, but the push test failed. Is it flaky?
```
	 35 - launch_pc_commands_ros_feedback.launch.py (Failed)
```
I see it passing on the PR one. Maybe rerun?

**Comment by andermi** [comment]
> Is it flaky?

yeah seems to be flaky. I ran the tests once and everything passed. I re-ran them and this one failed with a timeout. On one of the tests I fixed a timeout by adding a sleep at the end after rclcpp shutdown and then `launch_testing` picked up that the process ended. I was going to try the same with this one.

**Comment by mabelzhang** [comment]
Sounds promising

**Comment by andermi** [comment]
seems good?

**Comment by andermi** [comment]
not sure how to remove the (fortress, galactic) required status

**Comment by quarkytale** [comment]
@andermi @mabelzhang it seems like we can either add a branch filter in the workflow or just skip the workflow run temporarily, see [here](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks). What would be better?

**Comment by andermi** [comment]
@quarkytale @mabelzhang well, I'm not sure why we have a check for fortress/galactic in this PR at all. Can we just get rid of that check? I can't find it in the workflow...

**Comment by mabelzhang** [comment]
Looks good now. Fortress test is not there anymore - did y'all do something to skip it?

Should we remove the extra rosbag things and the extra timeouts originally added for the default RMW, and retest to see if we can do without them?

**Comment by andermi** [comment]
@mabelzhang I stopped generating rosbags for the two tests I enabled them for. There are only 1 second sleeps in two files I added right before an exe exits. I needed those even with the rmw change to cyclonedds.

File: buoy_tests/worlds/TestMachine.sdf [hunk 2]
```diff
@@ -5,71 +5,125 @@
...
         <scene>scene</scene>
         <ambient_light>1.0 1.0 1.0</ambient_light>
         <background_color>0.8 0.8 0.8</background_color>
         <camera_pose>-6 0 6 0 0.5 0</camera_pose>
       </plugin>
-      <!-- World control -->
-      <plugin filename="WorldControl" name="World control">
-        <ignition-gui>
-          <title>World control</title>
+<!-- Plugins that add functionality to the scene -->
```
Comment (mabelzhang) [inline buoy_tests/worlds/TestMachine.sdf:29]:
```suggestion
      <!-- Plugins that add functionality to the scene -->
```

**Review by mabelzhang — APPROVED** [review]
Okay I've looked at all the files. One minor style comment.
Re timing - okay, sounds good. I see the rosbag being disabled.

I'm approving because I'm done for the week. But there are 2 followup tickets we need to do below.

This test does seem to be flaky. We can merge now because we know it is flaky, but we should open an issue ticket to track it, and so that we know it's flaky in the future. Either we can try in another PR to add more timeout, or disable it so that we're not always merging on red CI.
```
	 35 - launch_pc_commands_ros_feedback.launch.py (Failed)
```

Don't forget to change the github workflow line 31 back to `main` in a followup PR.

**Comment by andermi** [comment]
It seems to be failing more consistently, and the only error is a print of `FAIL`

I'll try to debug it a bit more on Monday

**Comment by quarkytale** [comment]
Looks like it passed

**Pull request merged**